plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/logging/QueryCommentAsContextAppender.java

```diff
@@ -0,0 +1,102 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.logging;
+
+import io.trino.plugin.jdbc.QueryExecutionConnectorSessionContextAppender;
+import io.trino.spi.connector.ConnectorSession;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class QueryCommentAsContextAppender
+        implements QueryExecutionConnectorSessionContextAppender
+{
+    private final static List<String> PREDEFINED_MATCHES = Arrays.stream(PredefinedValue.values()).map(PredefinedValue::getMatchCase).toList();
+    private final static List<String> PREDEFINED_VALUE_CODES = Arrays.stream(PredefinedValue.values()).map(PredefinedValue::getPredefinedValueCode).toList();
+    private final static Pattern VALIDATION_PATTERN = Pattern.compile("[\\w ,=]|" + String.join("|", PREDEFINED_MATCHES));
+    private final String commentFormat;
+
+    public QueryCommentAsContextAppender(String commentFormat)
+    {
+        this.commentFormat = requireNonNull(commentFormat, "commentFormat is null");
+        checkArgument(!commentFormat.isBlank(), "Empty format is not allowed");
+        Matcher matcher = VALIDATION_PATTERN.matcher(commentFormat);
+        checkArgument(
+                matcher.results()
+                        .map(result -> result.group())
+                        .collect(Collectors.joining())
+                        .equals(commentFormat),
+                "Incorrect format: %s. Format may consist of only letters, digits, underscores, commas, spaces and predefined values: %s", commentFormat, String.join(", ", PREDEFINED_VALUE_CODES));
+    }
+
+    public QueryCommentAsContextAppender()
+    {
+        this("");
+    }
+
+    @Override
+    public String appendContext(String query, ConnectorSession session)
+    {
+        String message = commentFormat;
+        for (PredefinedValue predefinedValue : PredefinedValue.values()) {
+            message = message.replaceAll(predefinedValue.getMatchCase(), predefinedValue.value(session));
+        }
+        return new StringBuilder()
+                .append(query)
+                .append("/*")
```

Comment:
Either:
- Check that the message does not contain "*/" inside.
- Use the "--" comment style every line
And add a test for that